code/game/gamemodes/miniantags/tourist/tourist_arrivals.dm

```diff
@@ -0,0 +1,152 @@
+	for(var/mob/living/player in GLOB.mob_list)
+		if(player.mind && player.stat != DEAD)
+			crew_count++
+			if(!player.mind)
+				continue
+			if(player.mind.special_role)
+				antag_count++
+				continue
+	max_antag = round(crew_count / 10) + 1
+
+/datum/event/tourist_arrivals/start()
+	// Let's just avoid trouble, sending people into those is probably bad.
+	if(GAMEMODE_IS_CULT || GAMEMODE_IS_WIZARD || GAMEMODE_IS_NUCLEAR)
+		var/datum/event_container/EC = SSevents.event_containers[EVENT_LEVEL_MODERATE]
+		EC.next_event_time = world.time + 1 MINUTES
+		message_admins("Tourist Arrivals roll canceled due to gamemode. Rolling another midround in 60 seconds.")
+		return
+	if(SSsecurity_level.get_current_level_as_number() >= SEC_LEVEL_GAMMA) // Who would send more people to somewhere that's not safe?
+		var/datum/event_container/EC = SSevents.event_containers[EVENT_LEVEL_MODERATE]
+		EC.next_event_time = world.time + 1 MINUTES
+		message_admins("Tourist Arrivals roll canceled due to heightened alert. Rolling another midround in 60 seconds.")
+		return
+
+	INVOKE_ASYNC(src, PROC_REF(spawn_arrivals))
+
+/datum/event/tourist_arrivals/proc/spawn_arrivals()
+	var/list/candidates = SSghost_spawns.poll_candidates("Do you want to play as a Tourist?", null, TRUE)
+	// We'll keep spawning new tourists until we hit the max_spawn cap of tourists.
+	while(max_spawn > 0 && length(candidates))
+		var/turf/picked_loc = pick(GLOB.latejoin)
+		// Taking a random player from the candidate list.
+		var/mob/P = pick_n_take(candidates)
+		max_spawn--
+		var/datum/tourist/T = pick(/datum/tourist/human,
+								/datum/tourist/unathi,
+								/datum/tourist/vulp,
+								/datum/tourist/ipc,
+								/datum/tourist/skrell,
+								/datum/tourist/grey,
+								/datum/tourist/nian)
+		if(!istype(P))
+			continue
+		var/mob/living/carbon/human/M = new T.tourist_species(picked_loc)
+		// Picking a random objective, as all objectives are a subtype of /objective/tourist.
+		var/obj_tourist = pick(subtypesof(/datum/objective/tourist/))
+		var/datum/objective/tourist/O = new obj_tourist()
+		// Handles outfit, account and other stuff.
+		M.ckey = P.ckey
+		M.dna.species.after_equip_job(null, M)
+		M.age = rand(21, 50)
+		if(prob(50))
+			M.change_gender(MALE)
+		else
+			M.change_gender(FEMALE)
+		set_appearance(M)
+		GLOB.data_core.manifest_inject(M)
+		M.equipOutfit(T.tourist_outfit)
+		// Rolls a 20% probability, checks if 3 tourists have been made into tot and check if there's space for a new tot!
+		// If any is false, we don't make a new tourist tot
+		if(prob(20) && tot_number < 3 && antag_count < max_antag)
+			tot_number++
+			M.mind.add_antag_datum(/datum/antagonist/traitor)
+
+		// If they're a tot, they don't get tourist objectives neither the tourist greeting!
+		if(M.mind.special_role != SPECIAL_ROLE_TRAITOR)
+			M.mind.add_mind_objective(O)
+			greeting(M)
+		sucess_run = TRUE
+		spawned_in++
+	if(sucess_run)
+		message_admins("Made: [tot_number] traitors.")
```

Comment:
you should specify that these traitors came from the tourist event. With the amount of logs I got as an admin, it can be really easy to miss the event announcement and be confused as to why a bunch of traitors just got added to the round.